servers/src/main/java/tachyon/master/block/journal/BlockInfoEntry.java

```diff
@@ -50,4 +51,23 @@ public long getLength() {
   public JournalEntryType getType() {
     return JournalEntryType.BLOCK_INFO;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
```

Comment:
@verif I took a look at more of our `equals` implementations and some have the optimization while others don't. Since it probably will help sometimes, we should probably just make the implementation uniform, I've opened a ticket here: https://tachyon.atlassian.net/browse/TACHYON-1205